src/main/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -252,12 +252,15 @@ private List<ItemStack> calculateSecondaryOutputs()
      */
     private void calculateTools()
     {
-        for(ItemStorage item : getCleanedInput())
+        final List<ItemStorage> cleaned = getCleanedInput();
+        for (int index = 0; index < cleaned.size(); index++)
         {
-            for(ItemStack result: getSecondaryOutputs())
+            final ItemStorage item = cleaned.get(index);
+            for (ItemStack result : getSecondaryOutputs())
             {
-                if(ItemStackUtils.compareItemStacksIgnoreStackSize(item.getItemStack(), result, false, true) && result.isDamageableItem())
+                if (ItemStackUtils.compareItemStacksIgnoreStackSize(item.getItemStack(), result, false, true) && result.isDamageableItem())
                 {
+                    cleaned.set(index, new ImmutableItemStorage(new ItemStorage(item.getItemStack(), item.getAmount(), true)));
```

Comment:
Is that the output item (which is slightly more damaged), or the input item (which has not been damaged by the crafting recipe yet)?

Reply:
It's the input item.  At this point it's just telling it that it should ignore existing damage on the item -- actually damaging it happens elsewhere.

Reply:
Ah, I was wondering about the damaging part of that, thank you!